packages/plugins/src/components/plugin-area/index.tsx

```diff
@@ -16,15 +16,27 @@ import isShallowEqual from '@wordpress/is-shallow-equal';
 import { PluginContextProvider } from '../plugin-context';
 import { PluginErrorBoundary } from '../plugin-error-boundary';
 import { getPlugins } from '../../api';
+import type { PluginContext } from '../plugin-context';
+import type { Plugin } from '../../api';
 
-const getPluginContext = memoize( ( icon, name ) => ( { icon, name } ) );
+const getPluginContext = memoize(
+	( icon: PluginContext[ 'icon' ], name: PluginContext[ 'name' ] ) => ( {
+		icon,
+		name,
+	} )
+);
+
+type PluginAreaProps = {
+	scope?: string;
+	onError?: ( name: Plugin[ 'name' ], error: Error ) => void;
+};
 
 /**
  * A component that renders all plugin fills in a hidden div.
  *
- * @param {Object}             props
- * @param {string|undefined}   props.scope
- * @param {Function|undefined} props.onError
+ * @param                                                    props
+ * @param {string}                                           props.scope
+ * @param {( name: Plugin[ 'name' ], error: Error ) => void} props.onError
```

Comment:
Oh, this syntax works – nice

Reply:
Hold on, this is a `tsx` file now – shouldn't the inline type `PluginAreaProps` listed below with the arguments suffice?

Reply:
@adamziel @tyxla You're 100% right, and I already did try to do just that (three times now, just to be sure)... but unfortunately the documentation linter is very unhappy with this file, and I couldn't get it to work. So I needed to copy the types here :( 

I spent enough time on it that I don't think it's worth trying to fix, but happy for one of you to take a look :) 

```sh
Error: Command failed with exit code 1: "/Users/noahallen/source/gutenberg/node_modules/.bin/docgen" packages/plugins/src/index.js --output packages/plugins/README.md --to-token --use-token "Autogenerated API docs" --ignore "/unstable|experimental/i"

Error: Could not find corresponding parameter token for documented parameter 'props.scope' in function 'PluginArea'.
  at makeError (/Users/noahallen/source/gutenberg/node_modules/execa/lib/error.js:59:11)
  at handlePromise (/Users/noahallen/source/gutenberg/node_modules/execa/index.js:114:26)
  at processTicksAndRejections (internal/process/task_queues.js:95:5)
  at async Transform.<anonymous> (/Users/noahallen/source/gutenberg/bin/api-docs/update-api-docs.js:219:5) {
shortMessage: 'Command failed with exit code 1: "/Users/noahallen/source/gutenberg/node_modules/.bin/docgen" packages/plugins/src/index.js --output packages/plugins/README.md --to-token --use-token "Autogenerated API docs" --ignore "/unstable|experimental/i"',
command: '"/Users/noahallen/source/gutenberg/node_modules/.bin/docgen" packages/plugins/src/index.js --output packages/plugins/README.md --to-token --use-token "Autogenerated API docs" --ignore "/unstable|experimental/i"',
exitCode: 1,
```

Reply:
see #49812

Reply:
Great catch